manager2/src/app/admin/projects/projects.component.ts

```diff
@@ -175,15 +175,15 @@ export class ProjectsComponent implements OnInit {
         this.projectService.add({
             'uuid': this.new_project.uuid,
             'id': this.new_project.id,
+            'expire': this.new_project.expire,
```

Comment:
I 'think' you are going to need to convert date several times (in this file, but also on the user project file).

The issue is that angular datepicker give you a Date (a js entity type). But we don't want to store it in DB as it is, we want to store the unix time

(Using the `new Date(this.new_project.expire).getTime()` there was previously).
But then, you will need to convert it back to display it on the admin side (using something like [this](https://github.com/genouest/genouestaccountmanager/blob/364e137b3138e132539f3965618286ed0d179332/manager2/src/app/admin/project/project.component.ts#L190))

To summarize, the process would be:

* User select date in the field, and click send
* Before sending to the service, you convert the value using `new Date(this.new_project.expire).getTime()` (probably should not edit the variable in-place. Might be better to pass a whole new project Object with the values you want)
* Store the project in DB as you already do.
* When an admin access the projects admin page, get the project from DB, and convert it back to date from display

Reply:
It should be good now.

Reply:
Still need to convert it to unix time on the 'request project' side, and convert it back to JS Date when displaying admin-side

Reply:
Right, sorry for taking so long, I had to figure out what the heck a spread operator is.
Is there anything else I missed?

Reply:
Hmm, I think you still need to convert unix time to JS date on the admin side when getting the pending projets
Now that I'm thinking about it, you'll also need to manage the case when pending projects do not have an expire data (since current pending projects in production will not have one)

Also need to rollback package-lock.js and dev_quickstart/default.json I think?

Reply:
Ok so Angular's [(ngModel)] directive handles conversion between Unix timestamps and date objects automatically so there is no need to do admin-side conversions to JS date types but we do have to convert back to unix timestamps manually.
Also, the GitLens extention for VS Code is fantastic. Here you are working on the same problem 5 years ago : https://github.com/genouest/genouestaccountmanager/pull/33

Reply:
Ok, @mboudet, everything works now. The stupid html date input controls only accept a formated string (which is the original reason why the date_convert() function was written), which was causing me issues.

If there is a pending project with no expiration date, it will not cause any bugs but the admin accepting the project will have to enter one manually.